Update environment variable docs. (#5706)
vercel/turbo: docs/pages/repo/docs/handbook/environment-variables.mdx

import { Tabs, Tab } from '../../../../components/Tabs';

# Using Environment Variables

> Because environment variables are not captured in source code they're not as easily shared across machines. To set up environment variables for your repository you will need to take advantage of a number of Turborepo features.

Reviewer: ```suggestion
Because environment variables are not captured in source code, they're not as easily shared across machines. To set up environment variables for your repository you will need to take advantage of a number of Turborepo features.
```

Reviewer: Also "advantage of a number of Turborepo features" is a little awkward and seems to emphasize how many damn features we have to handle env vars (which tbf is a lot)

Author: I mean, I _think_ we have precisely the minimum number of environment variable features required in order to account for all use cases. 😅

Necessary complexity, and all that jazz.

## Working with Next.js and other libraries that load environment variables

> If your framework automatically loads environment variables from a particular file for you, you must make sure that `turbo` knows where that framework is going to load them from. This shows a baseline configuration for Next.js and Vite, who load their environment files themselves:

Reviewer: ```suggestion
If your framework automatically loads environment variables from a particular file, such as `.env`, you must indicate to `turbo` the location of these files. This shows a baseline configuration for Next.js and Vite, who load their environment files themselves:
```

<Tabs items={['Next.js', 'Vite']} storageKey="selected-framework">
  <Tab>
    ```jsonc
    {
      "$schema": "https://turbo.build/schema.json",
      "globalDotEnv": [".env"],
      "pipeline": {
        "build": {
          "dotEnv": [".env.production.local", ".env.local", ".env.production", ".env"]
        },
        "dev": {
          "dotEnv": [".env.development.local", ".env.local", ".env.development", ".env"]
        },
        "test": {
          "dotEnv": [".env.test.local", ".env.test", ".env"]
        }
      }
    }
    ```
  </Tab>
  <Tab>
    ```jsonc
    {
      "$schema": "https://turbo.build/schema.json",
      "globalDotEnv": [".env"],
      "pipeline": {
        "build": {
          "dotEnv": [".env.production.local", ".env.production", ".env.local", ".env"]
        },
        "dev": {
          "dotEnv": [".env.development.local", ".env.development", ".env.local", ".env"]
        },
        "test": {
          "dotEnv": [".env.test.local", ".env.test", ".env.local", ".env"]
        }
      }
    }
    ```
  </Tab>
</Tabs>

## Loading Your Own Variables

> If you need to load a large number of environment variables into your environment just-prior to execution, we recommend using `dotenv-cli`. It's the simplest way to bring your environment variables into your development tasks.

Reviewer: Any reason `just-prior` is hyphenated?

Author: nope.

> Turborepo does not load any `.env` files into the environment! Your task must handle loading of the `.env` files itself.

Reviewer 2: I would use `<Callout>` here.

Author: It's a callout in the other spot I copied it from; since we're doing a walkthrough here it felt like gilding the lily. Will switch to callout.
